analyzer/java/com/engflow/bazel/invocation/analyzer/suggestionproviders/UseRemoteCachingSuggestionProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 EngFlow Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.engflow.bazel.invocation.analyzer.suggestionproviders;
+
+import com.engflow.bazel.invocation.analyzer.Suggestion;
+import com.engflow.bazel.invocation.analyzer.SuggestionCategory;
+import com.engflow.bazel.invocation.analyzer.SuggestionOutput;
+import com.engflow.bazel.invocation.analyzer.core.DataManager;
+import com.engflow.bazel.invocation.analyzer.core.MissingInputException;
+import com.engflow.bazel.invocation.analyzer.core.SuggestionProvider;
+import com.engflow.bazel.invocation.analyzer.dataproviders.remoteexecution.RemoteCachingUsed;
+import java.util.ArrayList;
+import java.util.List;
+
+/** A {@link SuggestionProvider} that suggests using remote caching, if not already used. */
+public class UseRemoteCachingSuggestionProvider extends SuggestionProviderBase {
+  private static final String ANALYZER_CLASSNAME =
+      UseRemoteCachingSuggestionProvider.class.getName();
+  private static final String SUGGESTION_ID_USE_REMOTE_CACHING = "UseRemoteCaching";
+
+  @Override
+  public SuggestionOutput getSuggestions(DataManager dataManager) {
+    try {
+      boolean remoteCachingUsed =
```

Comment:
I assume this is true if remote execution is used?

Reply:
It could be that we don't detect RC, but do detect RE.

We check for `remote action cache check` to detect RC, and for completely other events to detect RE.

Say we have an invocation where all targets are marked as not cacheable (via tag `no-cache` or `no-remote-cache`), but you use remote execution. Then a profile might not have any RC checks, but events indicating RE was used.

Is it possible to use RE without RC?
If not, should we not show this suggestion, if RE is used?

Should we add a suggestion for the above case, such as checking whether actions that were marked as not cacheable should be investigated?

Reply:
It is technically possible to use RE and disable RC, but it seems highly unusual. I would probably not show this recommendation if RE is used.